### ISSUE TITLE
Introduce `patrol develop` command with R button for restarting the entry point

### ISSUE THREAD

**Issue by mateuszwojtczak** [leancodepl/patrol#1030]

**Comment by bartekpacia** [comment]
See also #982

**Comment by bartekpacia** [comment]
Do we need to require developers to [complete the native setup](https://patrol.leancode.co/native/setup) to use `patrol develop`?

The main reasons to remove `patrol drive` in favor of `patrol test` were:
- inability to get reports in a common format like JUnit
- inability to use device farm platforms (like Firebase Test Lab)

none of which make sense for local development of tests.

This would bring back one of Patrol's initial design goals – requiring 0 to minimal native setup (we had that goal abandon when we moved to `patrol test`). Literally, no setup would be required: just `patrol develop -t integration_test/your_test.dart` and you get Hot Restart and native automation. 

We wouldn't use `flutter drive` - we don't want it anymore - but `flutter run` is enough for this case. Alternatively, we could maintain our own Dart VM service connection.

**Comment by mateuszwojtczak** [comment]
I totally agree with that. If this doesn't mean we have to put the double amount of work to achieve this - `patrol develop` completely assumes running locally and it's ok to NOT require native setup. However, there will be cases when users want to run `patrol develop` with their tests having native automation inside - what would happen then?

If we have to throw an exception, we should somehow tell the users (e.g. in the docs) about this limitation.

**Comment by bartekpacia** [comment]
Scope:
- [x] get the core functionality to work (`r`/`R` to restart app,  `q`/`Q` to quit app)
- [x] pipe logs from `flutter logs`
- [ ] don't print "error - execution interrupted" when the user does CTRL+C or Q

**Comment by bartekpacia** [comment]
> However, there will be cases when users want to run patrol develop with their tests having native automation inside - what would happen then?

What cases do you have in mind? I can't think of any but I probably forgot about something.
